api/src/services/assessmentsService.ts

```diff
@@ -403,7 +400,47 @@ export const constructParticipantAssessmentSummary = async (
   participantPrincipalId: number,
   programAssessmentId: number
 ): Promise<ParticipantAssessmentSubmissionsSummary> => {
-  return;
+  const [highestState] = await db('assessment_submissions')
+    .select('assessment_submission_states.title')
+    .join(
+      'assessment_submission_states',
+      'assessment_submission_states.id',
+      'assessment_submissions.assessment_submission_state_id'
+    )
+    .where('assessment_submissions.principal_id', participantPrincipalId)
+    .andWhere('assessment_submissions.assessment_id', programAssessmentId)
+    .orderBy('assessment_submissions.assessment_submission_state_id', 'desc')
+    .limit(1);
+
+  const [mostRecentSubmittedDate] = await db('assessment_submissions')
+    .select('submitted_at')
+    .where('principal_id', participantPrincipalId)
+    .andWhere('assessment_id', programAssessmentId)
+    .orderBy('submitted_at', 'desc')
+    .limit(1);
+
+  const totalNumSubmissions = await listParticipantProgramAssessmentSubmissions(
+    participantPrincipalId,
+    programAssessmentId
+  );
+
+  const [highestScore] = await db('assessment_submissions')
+    .select('score')
+    .where('principal_id', participantPrincipalId)
+    .andWhere('assessment_id', programAssessmentId)
+    .orderBy('score', 'desc')
+    .limit(1);
+
+  const participantAssessmentSummary: ParticipantAssessmentSubmissionsSummary =
+    {
+      principal_id: participantPrincipalId,
+      highest_state: highestState.title,
+      most_recent_submitted_date: mostRecentSubmittedDate.submitted_at,
+      total_num_submissions: totalNumSubmissions.length,
+      highest_score: highestScore.score,
+    };
+
+  return participantAssessmentSummary;
```

Comment:
I removed the null checks on these, since the assessment summary should always return a complete summary and should never return null.

That said, let's have a chat today to see what you think about default values for some of these that may indeed return null, zero, or some other incorrect data.